lang/csharp/src/apache/main/AvroDecimal.cs

```diff
@@ -0,0 +1,929 @@
+        {
+            return ToUInt64(value);
+        }
+
+        /// <summary>
+        /// Creates an unsigned long from a given <see cref="AvroDecimal"/>.
+        /// </summary>
+        /// <param name="value">The <see cref="AvroDecimal"/>.</param>
+        /// <returns>An unsigned long.</returns>
+        public static ulong ToUInt64(AvroDecimal value)
+        {
+            return value.ToType<ulong>();
+        }
+
+        public static explicit operator float(AvroDecimal value)
+        {
+            return ToSingle(value);
+        }
+
+        /// <summary>
+        /// Creates a double from a given <see cref="AvroDecimal"/>.
+        /// </summary>
+        /// <param name="value">The <see cref="AvroDecimal"/>.</param>
+        /// <returns>A double.</returns>
+        public static float ToSingle(AvroDecimal value)
+        {
+            return value.ToType<float>();
+        }
+
+        public static explicit operator double(AvroDecimal value)
+        {
+            return ToDouble(value);
+        }
+
+        /// <summary>
+        /// Creates a double from a given <see cref="AvroDecimal"/>.
+        /// </summary>
+        /// <param name="value">The <see cref="AvroDecimal"/>.</param>
+        /// <returns>A double.</returns>
+        public static double ToDouble(AvroDecimal value)
+        {
+            return value.ToType<double>();
+        }
+
+        public static explicit operator decimal(AvroDecimal value)
+        {
+            return ToDecimal(value);
+        }
+
+        /// <summary>
+        /// Creates a decimal from a given <see cref="AvroDecimal"/>.
+        /// </summary>
+        /// <param name="value">The <see cref="AvroDecimal"/>.</param>
+        /// <returns>A decimal.</returns>
+        public static decimal ToDecimal(AvroDecimal value)
+        {
+            return value.ToType<decimal>();
+        }
+
+        public static explicit operator BigInteger(AvroDecimal value)
+        {
+            return ToBigInteger(value);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="BigInteger"/> from a given <see cref="AvroDecimal"/>.
+        /// </summary>
+        /// <param name="value">The <see cref="AvroDecimal"/>.</param>
+        /// <returns>A <see cref="BigInteger"/>.</returns>
+        public static BigInteger ToBigInteger(AvroDecimal value)
+        {
+            var scaleDivisor = BigInteger.Pow(new BigInteger(10), value.Scale);
+            var scaledValue = BigInteger.Divide(value.UnscaledValue, scaleDivisor);
+            return scaledValue;
+        }
+
+        public static implicit operator AvroDecimal(byte value)
+        {
+            return FromByte(value);
+        }
+
+        /// <summary>
+        /// Creates an <see cref="AvroDecimal"/> from a byte.
+        /// </summary>
+        /// <param name="value">The byte.</param>
+        /// <returns>An <see cref="AvroDecimal"/>.</returns>
+        public static AvroDecimal FromByte(byte value)
```

Comment:
i think the same with the froms (just have the cast operators), unless there's a compelling reason to have them I can't think of. as a general rule, smaller api surface area is better.

Reply:
Same as above. If I remove them you get [CA2225](https://docs.microsoft.com/en-gb/visualstudio/code-quality/ca2225?view=vs-2019).

Happy to suppress?

Reply:
up to you i think.